### ISSUE TITLE
[cert.ci.jenkins.io] Service Principal used by `cert.ci.jenkins.io` to spawn Azure agents expires on `2024-08-24`

### ISSUE THREAD

**Issue by dduportal** [jenkins-infra/helpdesk#4218]
### Service(s)

cert.ci.jenkins.io

### Summary

The Azure client credentials used by cert.ci.jenkins.io to start Azure VM/Container agents (ref. ) expires on 24 August 2024.

We have to:
- Set up `updatecli` in the repository https://github.com/jenkins-infra/azure to generate automatically a PR to update tghis credential
- Rotate this credential using the new PR

⚠️ Once this PR is merged, the deployment of Azure agents on ci.jenkins.io will fail until the Jenkins credential `azure-jenkins-sponsorship-credentials` (managed manually) is updated - ref. https://github.com/jenkins-infra/jenkins-infra/blob/abf8a194da9247dc955f832162565d9c3322c4b3/hieradata/clients/controller.cert.ci.jenkins.io.yaml#L130

**Comment by dduportal** [comment]
Update:

- Added automation in https://github.com/jenkins-infra/azure/pull/796
- New PR opened for credential rotation: https://github.com/jenkins-infra/azure/pull/797

**Comment by dduportal** [comment]
Update:

- Credential updated and verified with success on cert.ci.jenkins.io
- Azure Cloud configuration verified with success on cert.ci.jenkins.io
- Controller restarted: the `Infra-Team-Validate-Agents` job did allocate VMs with success
